unstructured/partition/common.py

```diff
@@ -148,14 +137,15 @@ def normalize_layout_element(
             )
         else:
             return ListItem(
-                text=text if text else "",
+                text=text,
                 coordinates=coordinates,
                 coordinate_system=coordinate_system,
                 metadata=class_prob_metadata,
                 detection_origin=origin,
             )
 
     elif element_type in TYPE_TO_TEXT_ELEMENT_MAP:
+        assert isinstance(element_type, str)
```

Comment:
what is this for?

Reply:
The `element_type` variable can be `None` (set on L119 with `.get()` on dict with no default). `None` is not a valid key for `TYPE_TO_TEXT_ELEMENT_MAP` which is `dict[str, type[Text]]`. This causes the type checker to rightfully complain. However, all the keys of that dict are str and so if it passes the `elif element_type in TYPE_TO_TEXT_ELEMENT_MAP` test then we know it has to be a `str` value.

Adding this assert just communicates to the type-checker this knowledge that it's not sophisticated enough to work out for itself (which would be way more sophisticated than you want a type-checker to be).

I expect there are ways to modify that code so the type of `element_type` is `str` and not `Optional[str]` (Optional is often an unnecessary complication/expense), but I didn't want to get into that in this PR.